recipes_source/torch_logs.py

```diff
@@ -0,0 +1,76 @@
+"""
+(beta) Using TORCH_LOGS python API with torch.compile
+==========================================================================================
+**Author:** `Michael Lazos <https://github.com/mlazos>`_
+"""
+
+import logging
+
+######################################################################
+#
+# This tutorial introduces the TORCH_LOGS environment variable and python API and
+# demonstrates how to apply it to observe the phases torch.compile.
+#
+# .. note::
+#
+#   This tutorial requires PyTorch 2.2.0 or later.
+#
+#
+
+
+######################################################################
+# Setup
+# ~~~~~~~~~~~~~~~~~~~~~
+# In this example we'll setup a simple python function which performs an elementwise
+# add and observe the compilation process with TORCH_LOGS python API.
+#
+# .. note::
+#
+#   There is also an environment variable TORCH_LOGS, which can be used to
+#   change logging settings at the commandline. The equivalent environment
+#   variable setting is shown for each example.
+
+import torch
+
+
+@torch.compile()
+def fn(x, y):
+    z = x + y
+    return z + 2
+
+
+inputs = (torch.ones(2, 2, device="cuda"), torch.zeros(2, 2, device="cuda"))
+
+
+# print separator and reset dynamo
+# between each example
+def separator(name):
+    print(f"==================={name}=========================")
+    torch._dynamo.reset()
+
+
+separator("Dynamo Tracing")
+# View dynamo tracing
+# TORCH_LOGS="+dynamo"
+torch._logging.set_logs(dynamo=logging.DEBUG)
+fn(*inputs)
+
+separator("Traced Graph")
+# View traced graph
+# TORCH_LOGS="graph"
+torch._logging.set_logs(graph=True)
+fn(*inputs)
+
+separator("Fusion Decisions")
+# View fusion decisions
+# TORCH_LOGS="fusion"
+torch._logging.set_logs(fusion=True)
+fn(*inputs)
+
+separator("Output Code")
+# View output code generated by inductor
+# TORCH_LOGS="output_code"
+torch._logging.set_logs(output_code=True)
+fn(*inputs)
+
+separator("")
```

Comment:
Some sort of conclusion with links to more torch.compile / dynamo docs would be nice to have.